File: src/com/boardgamegeek/provider/GamesIdColorsNameProvider.java
package com.boardgamegeek.provider;

import android.net.Uri;

import com.boardgamegeek.provider.BggContract.GameColors;
import com.boardgamegeek.provider.BggContract.Games;
import com.boardgamegeek.provider.BggDatabase.Tables;
import com.boardgamegeek.util.SelectionBuilder;

public class GamesIdColorsNameProvider extends BaseProvider {

	@Override
	protected SelectionBuilder buildSimpleSelection(Uri uri) {
		int gameId = Games.getGameId(uri);
		String color = uri.getLastPathSegment();
		return new SelectionBuilder().table(Tables.GAME_COLORS).whereEquals(GameColors.GAME_ID, gameId)
				.whereEquals(GameColors.COLOR, color);
	}

	@Override
	protected String getPath() {
		return "games/#/colors/*";
	}

	@Override
	protected String getType(Uri uri) {
		return GameColors.CONTENT_ITEM_TYPE;
	}
}

File: app/src/main/java/com/boardgamegeek/provider/GamesIdColorsNameProvider.kt
package com.boardgamegeek.provider

import android.net.Uri
import com.boardgamegeek.provider.BggContract.*
import com.boardgamegeek.provider.BggContract.Companion.PATH_COLORS
import com.boardgamegeek.provider.BggContract.Companion.PATH_GAMES
import com.boardgamegeek.provider.BggDatabase.Tables

/**
 * /games/13/colors/green
 */
class GamesIdColorsNameProvider : BaseProvider() {
    override fun getType(uri: Uri) = GameColors.CONTENT_ITEM_TYPE

    override val path = "$PATH_GAMES/#/$PATH_COLORS/*"

    override fun buildSimpleSelection(uri: Uri): SelectionBuilder {
        val gameId = Games.getGameId(uri)
        val color = uri.lastPathSegment
        return SelectionBuilder()
            .table(Tables.GAME_COLORS)
            .whereEquals(GameColors.Columns.GAME_ID, gameId)
            .whereEquals(GameColors.Columns.COLOR, color)
    }
}
